### PR TITLE
Jolt tutorial

### PR DESCRIPTION
**Pull request by Roee-87**
This PR adds the Jolt tutorial code to the public `sindri-resources` repo. This tutorial will be referenced in the documentation for the Jolt "how-to" guide.  The documentation PR will follow approval of this PR (Note that links that reference that "how-to" guide are included but will not work right now.  Both PRs will be merged at the same time, but this PR will need to be reviewed first).

To test the circuit, navigate to sindri-resources/circuit_tutorials/jolt/sha3_chain
Change sample.env to .env and enter your Sindri API key.
Run:

cargo run --release

### CODE REVIEW

**Comment by Roee-87** [comment]
Please note that the following link that is included in the README in `circuit_tutorials/jolt/` https://sindri.app/docs/how-to-guides/frameworks/jolt/ will NOT work.  The documentation PR will be reviewed after this PR.

File: circuit_tutorials/jolt/sha3_chain/README.md
Comment (grahamu) [inline circuit_tutorials/jolt/sha3_chain/README.md:1]: Please use a spell-checker on textual documentation like this. There are numerous spelling/typo errors in this short document.

File: circuit_tutorials/jolt/sha3_chain/rustfmt.toml
Comment (grahamu) [inline circuit_tutorials/jolt/sha3_chain/rustfmt.toml:1]: Nice professional touch!

File: circuit_tutorials/jolt/sha3_chain/README.md
Comment (Roee-87) [reply circuit_tutorials/jolt/sha3_chain/README.md:1]: done [f551793](https://github.com/Sindri-Labs/sindri-resources/pull/100/commits/f5517935de42e1d18e0909b73b92ffdfb9c9fc76) and [1cba93a](https://github.com/Sindri-Labs/sindri-resources/pull/100/commits/1cba93af03b4617e16ee8fdd28e00edd4c1e531b)

(I reinstalled the vscode checker extension.)

Comment (grahamu) [reply circuit_tutorials/jolt/sha3_chain/README.md:1]: much better thanks

File: circuit_tutorials/jolt/sha3_chain/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+// This struct stores the bytecode and memory_init data used to recreate the
+// preprocessing struct.
+#[derive(Serialize, Deserialize)]
+pub struct PreprocessingStruct {
+    pub bytecode: Vec<ELFInstruction>,
+    pub memory_init: Vec<(u64, u8)>,
+}
+
+// This function is deserializes the base64 encoded proof data and returns the
```
Comment (grahamu) [inline circuit_tutorials/jolt/sha3_chain/src/utils.rs:41]: remove "is"

**Review by grahamu — COMMENTED** [review]
This works for me locally. After the pending removal of an errant "is" in a utils.rs docstring, this is ready to go.

File: circuit_tutorials/jolt/sha3_chain/src/utils.rs (hunk 1 above)
Comment (Roee-87) [reply circuit_tutorials/jolt/sha3_chain/src/utils.rs]: done [ca75ce8](https://github.com/Sindri-Labs/sindri-resources/pull/100/commits/ca75ce80870c5518d57e3bb6070b2a2a5ee54565)

File: circuit_tutorials/jolt/sha3_chain/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+use serde::{Deserialize, Serialize};
+
+use common::rv_trace::ELFInstruction;
+use jolt::{
+    field::JoltField,
+    jolt::vm::{rv32i_vm::RV32IJoltProof, JoltCommitments},
+    poly::commitment::commitment_scheme::CommitmentScheme,
+};
+
+// This struct is used to store the base64 encoded proof data generated by
```
Comment (grahamu) [inline circuit_tutorials/jolt/sha3_chain/src/utils.rs:12]:
```suggestion
// This struct stores the base64 encoded proof data generated by
```

**Review by grahamu — APPROVED** [review]
LGTM

**Review by nickpann — APPROVED** [review]
LGTM

**Pull request merged**